### PR TITLE
Pocket New Tab should display "Save to Pocket" for Hero and List components

### PR DESCRIPTION
**Pull request by JeaneC**
Edit: New animations further down (change to Fade instead of Rotate)
~~*Look at last commit only until I rebase this after b_1540842.~~

Adding the DSContextFooter to the Hero and List components.

Animations
Hero - https://drive.google.com/open?id=1JCpnuzHaLFUJNTvyjfRkvCUFXQALk8ci
List - https://drive.google.com/open?id=1kMHnDZ0zIRczyF4dkcJeVOcbWIR7XV-Y

List, Hero
Other Lists

### CODE REVIEW

**Comment by JeaneC** [comment]
@wolasi initial thoughts?

**Comment by wolasi** [comment]
Looking good!

**Comment by JeaneC** [comment]
CSS NPM Package is finally in 😅.

@wolasi From #5196 we have the same fade animation
Hero - https://drive.google.com/open?id=13OLAkYK308tUTr3iOXD7Brq21FJB3FzD
List - https://drive.google.com/file/d/16ccUntKJ_jPb7HZDyOzgLjOZBLWDCsFC/
Hero w/ View Count - https://drive.google.com/open?id=1JCEz5iiPT02WWJ1bcYzzy8cUts8x0SHF

Fixed the spacing you mentioned before
Spacing

**Comment by wolasi** [comment]
Looking good!

**Comment by punamdahiya** [comment]
@wolasi Just double checking as this patch will change domain name position and sponsored by label across all layouts where we show list items including numbered list
Some screenshots below for comparison:

In nightly:
listitem-old
list-spoc-old


With patch:
listitem-new
list-context-spoc

File: content-src/components/DiscoveryStreamComponents/List/List.jsx [hunk 1]
```diff
@@ -64,27 +65,23 @@ export class ListItem extends React.PureComponent {
...
               {this.props.excerpt && (
                 <div className="ds-list-item-excerpt clamp">
                   {this.props.excerpt}
                 </div>
               )}
             </div>
-            <p>
-              {this.props.context && (
-                <span>
-                  <span className="ds-list-item-context clamp">
```
Comment (punamdahiya) [inline content-src/components/DiscoveryStreamComponents/List/List.jsx]: noticed we are losing out on clamp which is useful if we have longer text, should be safe to add it in DSContextFooter

File: test/unit/content-src/components/DiscoveryStreamComponents/List.test.jsx [hunk 2]
```diff
@@ -152,20 +145,6 @@ describe("<ListItem> presentation component", () => {
...
-    assert.lengthOf(contextEl, 0);
-  });
-
-  it("should contain 'span.ds-list-item-context' spoc element", () => {
-    const wrapper = shallow(<ListItem {...ValidLSpocListItemProps} />);
-
-    const contextEl = wrapper.find("span.ds-list-item-context");
-    assert.lengthOf(contextEl, 1);
-  });
-
```
Comment (punamdahiya) [inline test/unit/content-src/components/DiscoveryStreamComponents/List.test.jsx]: We should bring this test in DSContextFooter essentially checking if we are setting story-sponsored-label when context is provided

**Comment by punamdahiya** [comment]
We need to fix hero card to not display context messaging for spoc item.  @wolasi including for design review of how sponsored by label show for hero record. 

spoc-hero

**Comment by wolasi** [comment]
> @wolasi Just double checking as this patch will change domain name position and sponsored by label across all layouts where we show list items including numbered list
> Some screenshots below for comparison:
> 
> In nightly:
> listitem-old
> list-spoc-old
> 
> With patch:
> listitem-new
> list-context-spoc

Looks good! 👍

**Comment by wolasi** [comment]
> We need to fix hero card to not display context messaging for spoc item. @wolasi including for design review of how sponsored by label show for hero record.
> 
> spoc-hero

Sponsored label should still be at the bottom for a sponsored primary story in a hero. Only the source should be above the title. The text size and color of the sponsored label are in correct. Color should be grey 50 and font-size should be 13px, line-height 24px

**Comment by JeaneC** [comment]
> 
> Sponsored label should still be at the bottom for a sponsored primary story in a hero. Only the source should be above the title. The text size and color of the sponsored label are in correct. Color should be grey 50 and font-size should be 13px, line-height 24px

Sounds good. We can just then fully reuse the same Footer component from List and Card.

Screen Shot 2019-08-08 at 9 43 37 AM

**Comment by wolasi** [comment]
> > Sponsored label should still be at the bottom for a sponsored primary story in a hero. Only the source should be above the title. The text size and color of the sponsored label are in correct. Color should be grey 50 and font-size should be 13px, line-height 24px
> 
> Sounds good. We can just then fully reuse the same Footer component from List and Card.
> 
> Screen Shot 2019-08-08 at 9 43 37 AM

👍

File: test/unit/content-src/components/DiscoveryStreamComponents/List.test.jsx (hunk 2 above)
Comment (JeaneC) [reply test/unit/content-src/components/DiscoveryStreamComponents/List.test.jsx]: Spoke about keeping the test here and testing for SPOC in the DSContextFooter in List.

Upgraded test cases for Hero and DSCard to also test for Spoc!

File: test/unit/content-src/components/DiscoveryStreamComponents/List.test.jsx [hunk 3]
```diff
@@ -125,6 +129,15 @@ describe("<ListItem> presentation component", () => {
     title: "FAKE_TITLE",
     domain: "example.com",
     image_src: "FAKE_IMAGE_SRC",
+    context_type: "pocket",
+  };
+  const ValidLSpocListItemProps = {
```
Comment (punamdahiya) [inline test/unit/content-src/components/DiscoveryStreamComponents/List.test.jsx]: nit: typo ValidLSpocListItemProps

File: content-src/components/DiscoveryStreamComponents/DSContextFooter/DSContextFooter.jsx [hunk 4]
```diff
@@ -26,7 +26,7 @@ export class DSContextFooter extends React.PureComponent {
 
     return (
       <div className="story-footer">
-        {context && <p className="story-sponsored-label">{context}</p>}
+        {context && <p className="story-sponsored-label clamp">{context}</p>}
```
Comment (punamdahiya) [inline content-src/components/DiscoveryStreamComponents/DSContextFooter/DSContextFooter.jsx]: clamp by itself here doesn't work, should set -webkit-line-clamp: 1 for story-sponsored-label

File: content-src/components/DiscoveryStreamComponents/DSContextFooter/DSContextFooter.jsx [hunk 5]
```diff
@@ -9,13 +9,13 @@ import React from "react";
...
 
-const StatusMessage = ({ icon, fluentID }) => (
+export const StatusMessage = ({ icon, fluentID }) => (
   <div className="status-message">
     <span
       aria-haspopup="true"
       className={`story-badge-icon icon icon-${icon}`}
     />
-    <div className="story-context-label" data-l10n-id={fluentID} />
+    <div className="story-context-label clamp" data-l10n-id={fluentID} />
```
Comment (punamdahiya) [inline content-src/components/DiscoveryStreamComponents/DSContextFooter/DSContextFooter.jsx]: Probably safe to remove clamp here, to make it work we need -webkit-line-clamp same as below

**Comment by JeaneC** [comment]
Taskcluster failures aren't related to these changes. (reducers.jsm && ASRouter.jsm are not in this patch)

**Comment by Mardak** [comment]
Is there a bug associated to these changes?

**Comment by JeaneC** [comment]
> Is there a bug associated to these changes? Probably https://bugzilla.mozilla.org/show_bug.cgi?id=1569573 ?

Ah yes. Thanks for changing it to the right title 😅

**Pull request merged**